### PR TITLE
Create Circle class with necessary collisions

### LINKED ISSUE #1: Add Circle
Add shape `Circle`.

- [ ] Add shape `Circle`, using center position and radius
- [ ] Implments:
  - [ ] Equals, HashCode
  - [ ] Clone
  - [ ] Check two circles are sharing center
  - [ ] Intersections
    - [ ] `Line` and `Circle`
    - [ ] `LineSegment` and `Circle`
    - [ ] `Circle` and `Circle` (maybe easiest one)

FYI: There are good articles about intersection of line and circle[1](http://mathworld.wolfram.com/Circle-LineIntersection.html) [2](https://stackoverflow.com/questions/1073336/circle-line-segment-collision-detection-algorithm)

### PR DESCRIPTION
**Pull request by asmarton**
Closes #1

### CODE REVIEW

**Comment by 20chan** [comment]
This is really great. Thanks for this PR!

It looks like the PR contains unrelated formatting changes and also there are some code style differences, whitespace after type casting and whitespace after collection initializer `{ }`.
Would it be possible to revert those and format code?

File: cirno/Geometry/Intersections.cs [hunk 1]
```diff
@@ -65,15 +69,155 @@ internal static class Intersections {
...
+
             if (bSeg) {
                 if (u < 0 || 1 < u) {
                     intersects = default;
                     return false;
                 }
             }
 
-            intersects = new[] { new Vector(xNume / deno, yNume / deno) };
+            intersects = new[] {new Vector(xNume / deno, yNume / deno)};
```
Comment (20chan) [inline cirno/Geometry/Intersections.cs]: Could this be reverted?

File: cirno/Geometry/Intersections.cs [hunk 2]
```diff
@@ -65,15 +69,155 @@ internal static class Intersections {
                     return false;
                 }
             }
+
```
Comment (20chan) [inline cirno/Geometry/Intersections.cs]: Could this be reverted?

File: cirno/Geometry/Intersections.cs (hunk 1 above)
Comment (asmarton) [reply cirno/Geometry/Intersections.cs]: Fixed in 88b0ff3

File: cirno/Geometry/Intersections.cs (hunk 2 above)
Comment (asmarton) [reply cirno/Geometry/Intersections.cs]: Fixed in 88b0ff3

**Comment by 20chan** [comment]
I love it, thanks a lot!!

**Pull request merged**